main.go

```diff
@@ -0,0 +1,341 @@
+	}
+
+	// This set the sync period to 5m
+	syncPeriod := time.Duration(muocfg.SyncPeriodDefault)
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Namespace:              operatorNS,
+		Scheme:                 scheme,
+		MetricsBindAddress:     fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "312e6264.managed.openshift.io",
+		SyncPeriod:             &syncPeriod,
+		NewClient: func(_ cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+			return client.New(config, options)
+		},
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// Add UpgradeConfig controller to the manager
+	if err = (&upgradeconfig.ReconcileUpgradeConfig{
+		Client:                 mgr.GetClient(),
+		Scheme:                 mgr.GetScheme(),
+		MetricsClientBuilder:   metrics.NewBuilder(),
+		ClusterUpgraderBuilder: cub.NewBuilder(),
+		ValidationBuilder:      validation.NewBuilder(),
+		ConfigManagerBuilder:   configmanager.NewBuilder(),
+		Scheduler:              scheduler.NewScheduler(),
+		CvClientBuilder:        cv.NewBuilder(),
+		EventManagerBuilder:    eventmanager.NewBuilder(),
+		UcMgrBuilder:           ucm.NewBuilder(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "UpgradeConfig")
+		os.Exit(1)
+	}
+
+	// Add NodeKeeper controller to the manager
+	if err = (&nodekeeper.ReconcileNodeKeeper{
+		Client:                      mgr.GetClient(),
+		Scheme:                      mgr.GetScheme(),
+		ConfigManagerBuilder:        configmanager.NewBuilder(),
+		Machinery:                   machinery.NewMachinery(),
+		MetricsClientBuilder:        metrics.NewBuilder(),
+		DrainstrategyBuilder:        drain.NewBuilder(),
+		UpgradeConfigManagerBuilder: upgradeconfigmanager.NewBuilder(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "NodeKeeper")
+		os.Exit(1)
+	}
+
+	// Add MachineConfigPool controller to the manager
+	if err = (&machineconfigpool.ReconcileMachineConfigPool{
+		Client:                      mgr.GetClient(),
+		Scheme:                      mgr.GetScheme(),
+		UpgradeConfigManagerBuilder: ucm.NewBuilder(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "MachineConfigPool")
+		os.Exit(1)
+	}
+
+	ctx := context.TODO()
+
+	// Get a config to talk to the apiserver
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	// Add the Metrics Service and ServiceMonitor
+	if err := addMetrics(ctx, cfg); err != nil {
+		log.Error(err, "Metrics service is not added.")
+		os.Exit(1)
+	}
+
+	// Add the Custom Metrics Service
+	metricsClient, err := client.New(cfg, client.Options{})
```

Comment:
This is the other place where it causes issues due to a lack of type scheme registration.
I'm currently testing by passing a `mgr.GetClient()` into the `NewUpgradeCollector` instead and I think it is working okay (so far)

Reply:
Done :+1: